### PR TITLE
Fixes NPE in io.InputStream when the wrapped process throws an Exception

### PR DESCRIPTION
**Pull request by lukiano**
Also, subsequent reads after close() is called will return -1

### CODE REVIEW

File: src/main/scala/scalaz/stream/io.scala [hunk 1]
```diff
@@ -293,10 +297,12 @@ object io {
     @tailrec
     override def close() {
       if (cur.isHalt && chunks.isEmpty) {
-        chunks = null
+        chunks = Nil
```
Comment (djspiewak) [inline src/main/scala/scalaz/stream/io.scala]: Much more reasonable. I had a specific idea behind `null`, but `Nil` serves the purpose much better.

**Comment by djspiewak** [comment]
:+1:

**Comment by jedws** [comment]
@djspiewak @pchlupacek @pchiusano any reason this can't be merged?

**Comment by djspiewak** [comment]
I'm all in favor of merging it, I just wanted to wait for a second sign-off before I did.

**Pull request merged**